File: bmh_parent/bmh_rope_web2/src/main/webapp/js/service/systemRoleService.js
//服务层
app.service('systemRoleService',function($http){


	this.updateStatus=function(status,systemRoleId){
		return $http.get('../../systemRole/updateStatus/'+status+"/"+systemRoleId);
	}
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../../systemRole/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../../systemRole/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../../systemRole/findOne/'+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../../systemRole/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../../systemRole/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../../systemRole/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../../systemRole/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: bmh_parent/bmh_rope_interface/src/main/java/com/bmh/rope/service/SystemRoleService.java
package com.bmh.rope.service;

import com.bmh.pojo.Role;
import entity.PageResult;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SystemRoleService {

    /**
     * 返回全部列表
     * @return
     */
    public List<Role> findAll();


    /**
     * 返回分页列表
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);


    /**
     * 增加
     */
    public void add(Role systemRole);


    /**
     * 修改
     */
    public void update(Role systemRole);


    /**
     * 根据ID获取实体
     * @param id
     * @return
     */
    public Role findOne(String id);


    /**
     * 批量删除
     * @param ids
     */
    public void delete(String[] ids);

    /**
     * 分页
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(Role systemRole, int pageNum, int pageSize);

    void updateStatus(String status, String systemRoleId);

    Role findByUserName(String username);
}
